File: ProiectCS/ProiectCS/repository/FlightRepository.cs
using System;
using System.Collections;
using System.Collections.Generic;
using NET.model;

namespace NET.repository
{
    public interface FlightRepository : IRepository<Flight,int>
    {
        IEnumerable<Flight> filterFlightByDestinationAndDeparture(string destination, DateTime departure);
        IEnumerable<Flight> filterFlightByAvailableSeats();
    }
}

File: Java/TravelPersistence/src/main/java/chat/persistence/FlightRepository.java
package chat.persistence;


import travel.model.Flight;

import java.time.LocalDate;
import java.util.Collection;

public interface FlightRepository extends Repository<Flight , Integer>{
    Collection<Flight> filterFlightByDestinationAndDeparture(String destination, LocalDate departure);
    Collection<Flight> filterFlightByAvailableSeats();
}
